tests/psa-client-server/psasim/README.md

```diff
@@ -1,61 +1,45 @@
 # psasim
 
-This tool simulates a PSA Firmware Framework implementation.
-It allows you to develop secure partitions and their clients on a desktop computer.
-It should be able to run on all systems that support POSIX and System V IPC:
-e.g. macOS, Linux, FreeBSD, and perhaps Windows 10 WSL2.
+PSASIM holds necessary C source and header files which allows to test Mbed TLS in a "pure crypto client" scenario, i.e `MBEDTLS_PSA_CRYPTO_CLIENT && !MBEDTLS_PSA_CRYPTO_C`.
+In practical terms it means that this allow to build PSASIM with Mbed TLS sources and get 2 Linux applications, a client and a server, which are connected through Linux's shared memeory, and in which the client relies on the server to perform all PSA Crypto operations.
 
-Please note that the code in this directory is maintained by the Mbed TLS / PSA Crypto project solely for the purpose of testing the use of Mbed TLS with client/service separation. We do not recommend using this code for any other purpose. In particular:
+The goal of PSASIM is _not_ to provide a ready-to-use solution for anyone looking to implement the pure crypto client structure (see [Limitations](#limitations) for details), but to prove that it is possible to build and use Mbed TLS with a clean separation between the client and the server.
 
-* This simulator is not intended to pass or demonstrate compliance.
-* This code is only intended for simulation and does not have any security goals. It does not isolate services from clients.
+## Limitations
 
-## Building
+In current implementation:
 
-To build and run the test program make sure you have `make`, `python` and a
-C compiler installed and then enter the following commands:
+- Only Linux PC is supported.
+- There can be only 1 client connected to 1 server.
+- Shared memory is the only communication medium allowed. Others can be implemented (ex: net sockets), but in terms of simulation speed shared memory proved to be the fastest.
+- Server is not secure at all: keys and operation structs are stored on the RAM, so they can easily be dumped.
 
-```sh
-make run
-```
+## Testing
 
-Optionally the `DEBUG=1` command line option can be enabled to increase verbosity:
+For a quick and easy test of PSASIM it's strongly suggested to rely on the following 2 test components from `tests/scripts/components-psasim.sh`:
 
-```sh
-make DEBUG=1 run
-```
+- `component_test_psasim()`: builds the server and a couple of test clients which are used to evaluate some basic PSA Crypto API commands.
+- `component_test_suite_with_psasim()`: builds the server and _all_ the usual test suites (those found under the `<mbedtls-root>/tests/suites/*` folder) which are used by the CI and runs them. Only very few suites are skipped here (`test_suite_constant_time_hmac`,`test_suite_lmots`,`test_suite_lms`) , not because they cannot be run, but just because they take extremely long to execute and this is not nice for the CI.
```

Comment:
I think that segment could be reworded

> Only very few suites are skipped here (`test_suite_constant_time_hmac`,`test_suite_lmots`,`test_suite_lms`) , not because they cannot be run, but just because they take extremely long to execute and this is not nice for the CI.

A small subset of test suites (`test_suite_constant_time_hmac`,`test_suite_lmots`,`test_suite_lms`) are being skipped, for CI turnover time optimization. They can be run locally if required.